ext/ManoptManifoldsExt/manifold_functions.jl

```diff
@@ -18,7 +18,30 @@ the choice of typical distance in Matlab Manopt, the dimension of `M`. See
 [this note](https://github.com/NicolasBoumal/manopt/blob/97b6eb6b185334ab7b3991585ed2c044d69ee905/manopt/manifolds/fixedrank/fixedrankembeddedfactory.m#L76-L78)
 """
 function max_stepsize(M::FixedRankMatrices, p)
-    return manifold_dimension(M)
+    return max_stepsize(M)
+end
+max_stepsize(M::FixedRankMatrices) = manifold_dimension(M)
+
+function max_stepsize(M::Hyperrectangle, p)
```

Comment:
I have (still) not yet understood the hyper rectangle too much, but I believe you that this is correct. maybe we should stat documenting the max stepsize functions with formulae as well (then put them into the extension page)?

Reply:
Eventually we should have a custom line search for manifolds with corners that handles break points (positions when we hit a new boundary by following the descent direction with projection) but that's a somewhat bigger project. This `max_stepsize` is essentially a stop-gap solution that allows existing methods to sometimes work well enough. Right now `max_stepsize` defaults to 0 which makes line searches never work on `Hyperrectangle`. This is somewhat similar in spirit to `TangentBundle` where the global injectivity radius is also 0 but we have a specialized `max_stepsize`.

I can add a documentation about it somewhere.

Reply:
Thanks for the explanation, yes a bit of docs would be great especially since this is an interims fix.

Reply:
I've added docs regarding `max_stepsize`.